### PR TITLE
Improve scrolling on Geocharts

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA TNT-142

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by oleg-morozov** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6940/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8818/ : FAILURE in 10m 22s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1364/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6941/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1364/ : SUCCESS in 26m 49s

**Comment by oleg-morozov** [comment]
ok to test

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6953/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/136/ : SUCCESS in 7m 52s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6951/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8823/ : FAILURE in 10m 53s

**Comment by oleg-morozov** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6961/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8831/ : SUCCESS in 12m 18s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1367/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6962/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1367/ : SUCCESS in 36m 19s

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2282

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6979/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1119/ : SUCCESS in 15m 44s

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6983/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/140/ : SUCCESS in 7m 49s

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 😢 Some tests are failing. Consider fixing them.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|16.6620s|
|✅|**drilling.spec.ts**|2|-|26.1680s|
|✅|**dashboard.spec.ts**|9|-|71.1660s|
|❌|**[🎥 dateFilter.spec.ts](https://cing-dev.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/1//artifact/libs/sdk-ui-tests-e2e/cypress/videos/dateFilter.spec.ts.mp4)**|7|<span style="color: red">1</span>|22.8380s|
|✅|**attributeFilterButton.spec.ts**|2|-|15.7820s|

You can find full report [here](https://cing-dev.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/1/).

**Comment by oleg-morozov** [comment]
extended test

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboard.spec.ts**|9|-|58.9520s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|12.8380s|
|✅|**attributeFilterButton.spec.ts**|2|-|11.9400s|
|✅|**dateFilter.spec.ts**|7|-|18.7450s|
|✅|**drilling.spec.ts**|2|-|21.6350s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/147/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7045/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/147/ : SUCCESS in 10m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7042/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8897/ : SUCCESS in 11m 07s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2282

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7044/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1134/ : SUCCESS in 16m 00s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7043/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1386/ : SUCCESS in 26m 00s

File: libs/sdk-ui-geo/src/core/geoChart/GeoChartInner.tsx [hunk 1]
```diff
@@ -378,6 +380,7 @@ export class GeoChartInner extends React.PureComponent<IGeoChartInnerProps, IGeo
             onCenterPositionChanged,
             onZoomChanged,
             intl,
+            cooperativeGestures: cooperativeGestures as boolean,
```
Comment (tomassatinsky) [inline libs/sdk-ui-geo/src/core/geoChart/GeoChartInner.tsx:383]: why `as boolean`

Comment (tomassatinsky) [reply libs/sdk-ui-geo/src/core/geoChart/GeoChartInner.tsx:383]: !!cooperativeGestures

File: libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx [hunk 2]
```diff
@@ -239,6 +239,7 @@ export class PluggableGeoPushpinChart extends PluggableBaseChart {
 
         // keep height undef for AD; causes indigo-visualizations to pick default 100%
         const resultingHeight = this.environment === DASHBOARDS_ENVIRONMENT ? height : undefined;
+        const cooperativeGestures = this.environment === DASHBOARDS_ENVIRONMENT;
```
Comment (tomassatinsky) [inline libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx:242]: consider `!== ANALYTICAL_ENVIRONMENT`

**Comment by oleg-morozov** [comment]
extended test

**Comment by oleg-morozov** [comment]
extended check sonar

**Comment by oleg-morozov** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**attributeFilterButton.spec.ts**|9|-|34.3890s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|22.1250s|
|✅|**dashboard.spec.ts**|9|-|52.9590s|
|✅|**dateFilter.spec.ts**|7|-|14.2640s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|10.4320s|
|✅|**drilling.spec.ts**|2|-|17.9940s|
|✅|**dashboardFiltering.spec.ts**|2|-|22.6100s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|18.0140s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/150/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7084/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/150/ : SUCCESS in 9m 30s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-check-extended-sonar-PR2282

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7083/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1147/ : SUCCESS in 16m 24s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7082/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1395/ : UNSTABLE in 21m 52s

**Review by tomassatinsky — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2282

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2052/ : SUCCESS in 2m 08s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2335/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1149/ : SUCCESS in 14m 38s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8938/ : SUCCESS in 12m 54s

**Pull request merged**